precompiles/referenda/src/lib.rs

```diff
@@ -278,6 +275,21 @@ where
 		Ok(())
 	}
 
+	// Helper function for submit precompile functions
+	fn track_id_to_origin(track_id: u16) -> EvmResult<Box<OriginOf<Runtime>>> {
+		let origin: OriginOf<Runtime> = if track_id == 0 {
```

Comment:
How do we know track_id 0 is root ? That seems a very dangerous shortcut.

Reply:
I have not seen an OpenGov config with trackId 0 not corresponding to root origin. Moreover because it is a special origin, there is no other way to express this mapping.

Reply:
I'll add an integrity test to the runtimes but this is the best we can do.

Reply:
I think there should be a way to do that properly. It is too prone to error if one day we change the tracks

Reply:
@girazoki or @librelois to suggest some ideas ?

Reply:
Root is not part of the OpenGov origins, so whatever you do you have to map it manually somewhere.

To be sure of the match at compile time, we can create a public constant ROOT_TRACK_ID in a common crate and use it in `runtime/.../tracks.rs` and in the precompiles

Reply:
Why not make this part of the` From<GovOrigin>` instead of hardcoding? I will take a closer look at whether we can do something more clever, but at least this should probably be part of the runtime impl

Reply:
Taking a close look, cannot we use TracksInfo::tracks to build the impl?

Reply:
> Additionally we could add a check that the origin matches the track through `track_for` inside the precompile

I like this idea of using Tracks::tracks_for so I'm doing this now and will ping once ready for review.

Reply:
@girazoki @librelois I did a significant refactor to use Runtime::Tracks::tracks() in the precompile and runtime. This makes the TryFrom<u16> impl for CustomOrigin unnecessary but adds a TryFrom<String> impl for CustomOrigin.